namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+- `content` (string, required)
+
+## Distinct Element (Element)
+
+Abstract element representing an element that can be referenced or is a reference to another element.
+
+### Properties
+
+- `element` (Identifier, required) - identifier of this element's base element
+- `attributes`
+    - `id` (Identifier) - identifier of this element
+    - `ref` (Identifier) - reference to expanded element
+
+### Example
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+which is identical to:
+
+```json
+{
+    "element": {
+        "element": "id",
+        "content": "number"
+    },
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+and can be referred to as a _base element_:
+
+```json
+{
+    "element": "my-number"
+}
+```
+
+which _expands_ to:
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "ref": "my-number"
+    },
+    "content": 42
+}
+```
+
+## Select (Distinct Element)
+
+Element representing selection of options. Every item of content array represents one possible option.
+
+### Properties
+
+- `element`: select (string, required, fixed)
+- `content` (array[Option])
+
+## Option (Distinct Element)
+
+One choice in the selection.
+
+### Properties
+
+- `element`: option (string, required, fixed)
+- `content` (enum)
+    - (Distinct Element)
+    - (array[Distinct Element])
+
+#### Example
+
+```html
+<select>
+  <option>Volvo</option>
+  <option>Saab</option>
+</select>
+```
+
+```json
+{
+    "element": "select",
+    "content": [
+        {
+            "element": "option",
+            "content": [
+                {
+                    "element": "string",
+                    "content": "volvo"
+                }
+            ]
+        }
+        {
+            "element": "option",
+            "content": [
+                {
+                    "element": "string",
+                    "content": "Saab"
+                }
+            ]
+        }
+    ]
+}
+```
+
+# MSON DOM Elements Definition
+
+## MSON Element (Distinct Element)
+
+Base element for all MSON elements.
+
+The MSON Element adds attributes representing MSON _Type Definition_ and _Type Sections_. Note in MSON DOM _Nested Member Types_ _Type Section_ is the `content` of the element.
+
+### Properties
+
+- `attributes`
+    - `typeAttributes` (array) - _Type Definition_ attributes list, see _Type Attribute_  
+        - (enum[string])
+            - required
+            - optional
+            - fixed
+            - sample
+            - default
+    - `variable` (boolean) - Element content is _Variable Value_
+    - `sample` (MSON Element) - Alternative sample _Member Types_ value
+    - `default` (MSON Element) - Default value for _Member Types_
+    - `validation` - Not used, reserved for a future use
+    - `description` - Combined description of an MSON Element
+
+## Boolean Type (JSON:Boolean Type)
```

Comment:
Just a note, there are no primitives in the base spec anymore, as they were pulled out to the JSON namespace.